x/subscription/types/cronspec.go

```diff
@@ -415,13 +415,14 @@ func roundUpItem(spec CompiledCronItem, v uint32) uint32 {
 
 // CountPeriods counts how many datetimes between beginTime expirationTime matches the cron spec.
 // beginTime is not inclusive.
-func (spec CompiledCronSpec) CountPeriods(beginTime, expirationTime int64, tzoffset int32) (count uint64) {
+func (spec CompiledCronSpec) CountPeriods(beginTime, expirationTime int64, tzoffset int32, callback func()) (count uint64) {
 	count = 0
 	for {
 		beginTime = spec.RoundUp(beginTime+1, tzoffset)
 		if beginTime >= expirationTime {
 			break
 		}
+		callback()
```

Comment:
Is each RoundUp execution charge 1000 gas?
If yes, in your test case(1 yr period), how much will it be?

Reply:
```
>>> 365*24 * 1000 * 0.025
219000basecro
```
Not much, based on current gas price.

Reply:
The current SDK's fee model is subject to change: https://github.com/cosmos/cosmos-sdk/discussions/8224

Reply:
Isn't the RoundUp minute based?  Should it be 60 times more like 219000basecro x 60 = 0.1314CRO?

Reply:
Oh, with updated GasPerCollection, the gas is bigger:
```
>>>  365*24 * 31288 * 0.025
6852072basecro (0.685cro)
```

Reply:
ah, your test case is hourly.

Reply:
> ah, your test case is hourly.

yeah, I removed the minute part from the cron syntax, since it don't make much sense here.